### DISCUSSION TITLE
Type system

### DISCUSSION THREAD

**Discussion by jgarte** [phel-lang/phel-lang#281] [Ideas]
What do you think of exposing declared types in phel-lang that leverage php8's types?

Some inspiration:

https://github.com/hylang/hy/issues/2030
https://github.com/racket/typed-racket
https://github.com/pflanze/chj-schemelib
https://github.com/clojure/core.typed/wiki

**Comment by jenshaase** [comment]
Types are currently not a goal of Phel. A complete type system is a lot of work and is currently not doable in the current state of the project. 

However, It may also possible that someone else in the community can work on it Independently from the Phel team. As we can see with Clojures core.type or with PHP's Psalm and PHPStan, optionally type system can be implemented as a library.

**Comment by Chemaclass** [comment]
It might not be a full typed system, but now you can define your own structs and interfaces in phel, so it's somehow related to this topic. Check this out: https://phel-lang.org/documentation/interfaces/
